mne_connectivity/base.py

```diff
@@ -0,0 +1,548 @@
+                 n_epochs_used=None, **kwargs):
+        super().__init__(data, names=names, method=method,
+                         indices=indices, n_nodes=n_nodes,
+                         freqs=freqs, spec_method=spec_method,
+                         n_epochs_used=n_epochs_used, **kwargs)
+
+    def __repr__(self):  # noqa: D105
+        s = ", freq : [%f, %f]" % (self.freqs[0], self.freqs[-1])
+        s += ", nave : %d" % self.n_epochs_used
+        s += ', nodes, n_estimated : %d, %d' % (self.n_nodes,
+                                                self.n_estimated_nodes)
+        s += ', ~%s' % (sizeof_fmt(self._size),)
+        return "<SpectralConnectivity | %s>" % s
+
+
+@fill_doc
+class TemporalConnectivity(_Connectivity, TimeMixin):
+    """Temporal connectivity container.
+
+    Parameters
+    ----------
+    %(data)s
+    %(times)s
+    %(n_nodes)s
+    %(names)s
+    %(indices)s
+    %(method)s
+    %(n_epochs_used)s
+    """
+
+    def __init__(self, data, times, n_nodes, names=None, indices=None,
+                 method=None, n_epochs_used=None, **kwargs):
+        super().__init__(data, names=names, method=method,
+                         n_nodes=n_nodes, indices=indices,
+                         times=times, n_epochs_used=n_epochs_used,
+                         **kwargs)
+
+    def __repr__(self):  # noqa: D105
+        s = "time : [%f, %f]" % (self.times[0], self.times[-1])
+        s += ", nave : %d" % self.n_epochs_used
+        s += ', nodes, n_estimated : %d, %d' % (self.n_nodes,
+                                                self.n_estimated_nodes)
+        s += ', ~%s' % (sizeof_fmt(self._size),)
+        return "<TemporalConnectivity | %s>" % s
+
+
+@fill_doc
+class SpectroTemporalConnectivity(_Connectivity, SpectralMixin, TimeMixin):
+    """Spectrotemporal connectivity container.
+
+    Parameters
+    ----------
+    %(data)s
+    %(freqs)s
+    %(times)s
+    %(n_nodes)s
+    %(names)s
+    %(indices)s
+    %(method)s
+    %(spec_method)s
+    %(n_epochs_used)s
+    """
+
+    def __init__(self, data, freqs, times, n_nodes, names=None,
+                 indices=None, method=None,
+                 spec_method=None, n_epochs_used=None, **kwargs):
+        super().__init__(data, names=names, method=method, indices=indices,
+                         n_nodes=n_nodes, freqs=freqs,
+                         spec_method=spec_method, times=times,
+                         n_epochs_used=n_epochs_used, **kwargs)
+
+    def __repr__(self):  # noqa: D105
+        s = "time : [%f, %f]" % (self.times[0], self.times[-1])
+        s += ", freq : [%f, %f]" % (self.freqs[0], self.freqs[-1])
+        s += ", nave : %d" % self.n_epochs_used
+        s += ', nodes, n_estimated : %d, %d' % (self.n_nodes,
+                                                self.n_estimated_nodes)
+        s += ', ~%s' % (sizeof_fmt(self._size),)
+        return "<SpectroTemporalConnectivity | %s>" % s
```

Comment:
This seems like it could be DRYed out and put in the base class. If you make each subclass have (private?) class attributes like:
```
class _whateverbase:
    def __repr__(self):
        r = f'<{self.__class__.__name__} '
        if 'times' in self._dims:
            r += ...
        if 'freqs' in self._dims:
            r += ...
        ...
        r += '>'
        return r

class whatever(_whateverbase):
    _dims = ('times', 'freqs')

class whateverelse(_whateverbase):
    _dims = ('times')
```